bundles/org.eclipse.jface/src/org/eclipse/jface/dialogs/DialogSettings.java

```diff
@@ -87,6 +89,8 @@ public class DialogSettings implements IDialogSettings {
 
 	// A Map with all the keys mapped to array of strings.
 	private Map<String, String[]> arrayItems;
+	private Deque<String> lastFiveSearches;
+	private static final int SEARCHES_DISPLAYED = 5;
```

Comment:
I'd use something more descriptive and, in the case of the `Collection`, something that **doesn't** refer to its size (more future-proof):

```suggestion
	
	private Deque<String> filterHistory;
	
	private static final int FILTER_HISTORY_SIZE = 5;
```

Also, watch the line-spacing (preserve the existing spacing).

Reply:
done